docs/howto/build-guides/select-platforms.rst

```diff
@@ -0,0 +1,196 @@
+for both AMD64 and 64-bit RISC-V:
+
+.. code-block:: yaml
+
+   base: ubuntu@24.04
+   platforms:
+     amd64:
+       build-on: [amd64]
+       build-for: [amd64]
+     riscv64:
+       build-on: [riscv64]
+       build-for: [riscv64]
+
+Because the ``build-on`` and ``build-for`` entries are identical for each
+platform, the shorthand notation can be used instead:
+
+.. code-block:: yaml
+
+   base: ubuntu@24.04
+   platforms:
+     amd64:
+     riscv64:
+
+The results are the same with either snippet. Building on amd64 will
+produce one charm that runs on amd64. Building on ``riscv64`` will produce
+one charm that runs on ``riscv64``.
+
+.. note::
+
+    The ``build-for`` key may only contain one architecture, despite being a list.
+
+
+Select different architectures for building and running
+-------------------------------------------------------
+
+A charm may require cross-compilation to build. To create a charm for a
+different architecture, use the following ``charmcraft.yaml`` snippet:
+
+.. code-block:: yaml
+
+   base: ubuntu@24.04
+   platforms:
+     riscv64-cross:
+       build-on: [amd64]
+       build-for: [riscv64]
+
+Building on amd64 will produce one charm that runs on ``riscv64``.
+
+Note that the charm developer must ensure the charm is compatible with the
+target architectures. By default, the `charm`_,
+:ref:`python <craft_parts_python_plugin>`, :ref:`poetry <craft_parts_poetry_plugin>`,
+and :ref:`uv <craft_parts_uv_plugin>` plugins will install wheels for python packages
+for the ``build-on`` architecture rather than the ``build-for``. For more information,
+see `craft-parts#974`_.
+
+Select multiple OS releases
+---------------------------
+
+The resulting ``.charm`` file packed by charmcraft can only run on a single OS release
+or ``base``, using ``charmcraft.yaml`` nomenclature. A ``charmcraft.yaml`` can use
+multi-base syntax to create a set of charms, each for a different base. To do this, the
+base is defined in each platform entry instead of being defined with the
+top-level ``base`` and ``build-base`` keys.
+
+To build a charm for Ubuntu 22.04 and a charm for Ubuntu 24.04, use the
+following ``charmcraft.yaml`` snippet which uses :ref:`multi-base
+notation<reference-platforms-multi-base>`:
+
+.. code-block:: yaml
+
+   platforms:
+     ubuntu-22.04-amd64:
+       build-on: [ubuntu@22.04:amd64]
+       build-for: [ubuntu@22.04:amd64]
+     ubuntu-24.04-amd64:
+       build-on: [ubuntu@24.04:amd64]
+       build-for: [ubuntu@24.04:amd64]
+
+The ``build-on`` and ``build-for`` entries are identical for each platform, so
+the :ref:`multi-base shorthand notation
+<reference-platforms-multi-base-shorthand>` can be used:
+
+.. code-block:: yaml
+
+   platforms:
+     ubuntu@22.04:amd64:
+     ubuntu@24.04:amd64:
+
+With both snippets, building on amd64 will produce two charms, one for
+amd64 systems running Ubuntu 22.04 and one for amd64 systems running
+Ubuntu 24.04.
```

Comment:
```suggestion
With both snippets, building on AMD64 will produce two charms, one for
AMD64 hosts running Ubuntu 22.04 LTS and one for AMD64 hosts running
Ubuntu 24.04 LTS.
```

Since you changed it to "hosts" earlier in the page.

Reply:
I'm distinguishing here the build host from the running system, which is why I kept these as systems.

Reply:
Non-blocking, but what makes runtime different? Don't charms run on every kind of machine?